DigitalLearningSolutions.Data/Services/SectionContentService.cs

```diff
@@ -0,0 +1,65 @@
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using System.Data;
+    using System.Linq;
+    using Dapper;
+    using DigitalLearningSolutions.Data.Models.SectionContent;
+    using Microsoft.Extensions.Logging;
+
+    public interface ISectionContentService
+    {
+        SectionContent? GetSectionContent(int customisationId, int progressId, int sectionId);
+    }
+
+    public class SectionContentService : ISectionContentService
+    {
+        private readonly IDbConnection connection;
+        private readonly ILogger<SectionContentService> logger;
+
+        public SectionContentService(IDbConnection connection, ILogger<SectionContentService> logger)
+        {
+            this.connection = connection;
+            this.logger = logger;
+        }
+
+        public SectionContent? GetSectionContent(int customisationId, int progressId, int sectionId)
```

Comment:
One thing I've just noticed, this needs to take the candidate id rather than the progress id. As we should only return content if the candidate is enrolled on the course. If there is no progress record for the course then anything which relies on the progress record should be initialised to a sensible default value, e.g. 0 percent complete 🙂

Reply:
Can this be done using the query `GetOrCreateProgressId` frank made for course content? And if so would it be useful to extract it to a different helper method rather than have it within the course content service

Reply:
Yes, it should use that. It's not really a problem to have it in the course content service though, as it will be called from the controller. Does that make sense? So I won't expect that to be done in this PR 🙂